assets/src/editor/events/ticket-assignments-manager/ticket-assignments-manager-modal.js

```diff
@@ -659,201 +297,359 @@ class TicketAssignmentsManagerModal extends Component {
 	 */
-	getFormError = ( noAssignments ) => {
-		let formError = this.state.formError;
-		if ( noAssignments > 0 ) {
-			formError = this.dateCount === 1 ?
-				__(
-					'Event Dates must always have at least one Ticket' +
-					' assigned to them but one or more of the Event Dates' +
-					' below does not have any. Please correct the' +
-					' assignments for the highlighted cells.',
-					'event_espresso'
-				) : __(
-					'Tickets must always have at least one Event Date' +
-					' assigned to them but one or more of the Tickets' +
-					' below does not have any. Please correct the' +
-					' assignments for the highlighted cells.',
-					'event_espresso'
-				);
-		}
-		return formError ?
-			<FormInfo
-				formInfo={ formError }
-				dashicon={ 'warning' }
-				dismissable={ true }
-				colSize={ 10 }
-				offset={ 1 }
-			/> : null;
-	};
+	const ticketCell = useCallback(
+		( dateEntity, ticketEntity, dateTicketEntities ) => {
+			const {
+				hasTicket,
+				isAssigned,
+				isRemoved,
+			} = determineCurrentAssignment(
+				dateEntity,
+				ticketEntity,
+				dateTicketEntities,
+			);
+			const { currentlyAssigned } = getActionModifiers(
+				hasTicket,
+				isAssigned,
+				isRemoved,
+			);
+			const action = getAction( currentlyAssigned );
+			const assignmentsErrorClass = getAssignmentsErrorClass( dateEntity, ticketEntity );
+			const { icon, bgColor } = getIconAndBgColor(
+				hasTicket,
+				isAssigned,
+				isRemoved
+			);
+			return {
+				type: 'cell',
+				class: `ee-tam-date-row-ticket${ assignmentsErrorClass }`,
+				value: (
+					<IconButton
+						icon={ icon }
+						className={ bgColor }
+						size={ 45 }
+						onClick={ ( event ) => {
+							event.preventDefault();
+							event.stopPropagation();
+							action( dateEntity.id, ticketEntity.id );
+						} }
+						onKeyDown={ ( event ) => {
+							if ( event.keyCode === ENTER ) {
+								event.preventDefault();
+								event.stopPropagation();
+								action( dateEntity.id, ticketEntity.id );
+							}
+						} }
+					/>
+				),
+			};
+		},
+		[
+			assignedState,
+			getActionModifiers,
+			getAction,
+			getAssignmentsErrorClass,
+			getIconAndBgColor,
+			determineCurrentAssignment,
+		]
+	);
 
 	/**
 	 * @function
-	 * @param {Function} processChanges
-	 * @return {Object} rendered submit button
+	 * @param {Object} dateEntity
+	 * @return {Object} rendered table cell
+	 */
+	const dateHeader = useCallback(
+		( dateEntity ) => {
+			return {
+				type: 'cell',
+				class: 'ee-tam-date-label',
+				value: (
+					<div className="ee-tam-date-label-div">
+						<div className="ee-tam-date-id">
+							{ `#${ dateEntity.id }` }
+						</div>
+						<div className="ee-tam-date-label-text">
+							{ dateEntity.name }
+						</div>
+						<CalendarPageDate
+							startDate={ dateEntity.start }
+							statusClass={ getDateBgColorClass( dateEntity ) }
+							size={ 'small' }
+						/>
+					</div>
+				),
+			};
+		},
+		[]
```

Comment:
No dependencies needed because:

- `dateEntity` is passed in as an argument
- `getDateBgColorClass` and `CalendarPageDate` are both imports and thus will not change.